### PR TITLE
Fix buildbot

### PR DESCRIPTION
**Pull request by abergeron**
This fixes most of the problems on the buildbot.

Two problems remain:
 - test_use_scan_direct_output2 (theano.scan_module.tests.test_scan.T_Scan),
 - test_grad_multiple_outs_some_disconnected (theano.scan_module.tests.test_scan.T_Scan)

both in debug mode.  I'm looking at the first one, but it seems it's not a simple fix.  I haven't looked at the second one yet.

### CODE REVIEW

**Comment by abergeron** [comment]
I've fixed test_use_scan_direct_output2 which turned out to be a bug in get_scalar_constant_value()

File: theano/scan_module/tests/test_scan.py [hunk 1]
```diff
@@ -1772,8 +1777,16 @@ def reset_rng_fn(fn, *args):
         analytic_grad = reset_rng_grad_fn(v_u, v_x0, vW_in)
         utt.assert_allclose(analytic_grad[0][:2], numpy.zeros((2, 2)))
 
-    @attr('slow')
```
Comment (lamblin) [inline theano/scan_module/tests/test_scan.py]: I'm not sure what the threshold is to tag a test as "slow", so maybe we should keep it?

**Comment by lamblin** [comment]
I'm OK with the change to `filter_variable`, and the addition of `allow_convert`. Let's see how Travis reacta

File: theano/scan_module/tests/test_scan.py (hunk 1 above)
Comment (nouiz) [inline theano/scan_module/tests/test_scan.py]: 1s

**Comment by nouiz** [comment]
Documenting filter_variable would be great.

**Pull request merged**